catalog/dags/data_refresh/distributed_reindex.py

```diff
@@ -0,0 +1,425 @@
+
+
+class TempConnectionHTTPOperator(HttpOperator):
+    """
+    Wrapper around the HTTPOperator which allows templating of the conn_id,
+    in order to support using a temporary conn_id passed through XCOMs.
+    """
+
+    template_fields: Sequence[str] = (
+        "endpoint",
+        "data",
+        "headers",
+        # Extended to allow templating of conn_id
+        "http_conn_id",
+    )
+
+
+class TempConnectionHTTPSensor(HttpSensor):
+    """
+    Wrapper around the HTTPSensor which allows templating of the conn_id,
+    in order to support using a temporary conn_id passed through XCOMs.
+    """
+
+    template_fields: Sequence[str] = (
+        "endpoint",
+        "request_params",
+        "headers",
+        # Extended to allow templating of conn_id
+        "http_conn_id",
+    )
+
+
+def response_filter_status_check_endpoint(response: Response) -> str:
+    """
+    Handle the response from the `trigger_reindex` task.
+
+    This is used to get the status endpoint from the response, which is used to poll for the status
+    of the reindexing task.
+    """
+    status_check_url = response.json()["status_check"]
+    return urlparse(status_check_url).path
+
+
+def response_check_wait_for_completion(response: Response) -> bool:
+    """
+    Handle the response for `wait_for_reindex` Sensor.
+
+    Processes the response to determine whether the task can complete.
+    """
+    data = response.json()
+
+    if data["active"]:
+        # The reindex is still running. Poll again later.
+        return False
+
+    if data["error"]:
+        raise ValueError("An error was encountered during reindexing.")
+
+    logger.info(f"Reindexing done with {data['progress']}% completed.")
+    return True
+
+
+@task
+def get_worker_params(
+    estimated_record_count: int,
+    environment: str,
+    target_environment: Environment,
+    aws_conn_id: str = AWS_CONN_ID,
+):
+    """Determine the set of start/end indices to be passed to each indexer worker."""
+    # Defaults to one indexer worker in local development
+    worker_count = (
+        INDEXER_WORKER_COUNTS.get(target_environment)
+        if environment == PRODUCTION
+        else 1
+    )
+    records_per_worker = math.floor(estimated_record_count / worker_count)
+
+    return [
+        {
+            "start_id": worker_index * records_per_worker,
+            "end_id": (1 + worker_index) * records_per_worker,
+        }
+        for worker_index in range(worker_count)
+    ]
+
+
+@task
+def get_launch_template_version(
+    environment: str,
+    target_environment: Environment,
+    aws_conn_id: str = AWS_CONN_ID,
+):
+    """
+    Get the latest version of the launch template. Indexer workers will all be created with this
+    version. Importantly, this allows us to retry an individual indexer worker and ensure that
+    it will run with the same version of the indexer worker as the others, even if code has been
+    deployed to the indexer worker in the meantime.
```

Comment:
For posterity, might be worth mentioning that this retriability an intentional reason not to use the `$Latest` alias for launch template versions so that every request to use the launch template during this run will use the same launch template version.

This could say, "Get the latest version of the launch template at time of distributed reindex start", or something, to clarify that "latest" is relevant to a particular point in time. :thinking: 

Just a nit though, it might be over complicating it to try to explain it in this much detail, you've already covered the important bits.